syntax = "proto3";

option go_package = ".;proto";

import "google/protobuf/empty.proto";

service ShopCart {
    rpc AddGoodsToShopCart(ShopCartRequest)returns(google.protobuf.Empty);
    rpc QueryShopCart(UserInfoRequest)returns(ShopCartListResponse);
    rpc UpdateShopCart(ShopCartRequest)returns(google.protobuf.Empty);
    rpc DelGoodsInShopCart(ShopCartRequest)returns(google.protobuf.Empty);
}

message GoodsInfoRequest {
    int32 id = 1;
}

message UserInfoRequest {
    int32 id = 1;
}

message ShopCartListResponse {
    int32 total = 1;
    repeated ShopCartRequest data = 2;
}

message ShopCartRequest {
    int32 user_id = 1;
    int32 goods_id = 2;
    int32 nums = 3;
    bool checked = 4;
}
